### PR TITLE
Ask for the identity specifically if its not sent

### PR DESCRIPTION
**Pull request by mohitkhullar**
Signed-off-by: mohitkhullar <mkhullar1@bloomberg.net>

### CODE REVIEW

**Comment by mohitkhullar** [comment]
client side changes in bb cdb2api

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ cdb2api/cdb2api.c @@
+ Please port the change to cdb2jdbc if necessary.
+ If you add any internal configuration state to this file, please update the reset_the_configuration() function as well to include it.
@@ db/db_tunables.h @@
+ Please update the tunable documentation (docs/pages/config/config_files.md) accordingly.
@@ sqlite/ext/comdb2/tables.c @@
+ If a new system table is added, be sure to document it in docs/pages/programming/system_tables.md.
@@ sqlite/src/parse.y @@
+ Custom tokens and rules must be defined after SQLite's.
+ Please update SQL documentation and bubble diagrams to reflect the changes you've made, if necessary.
@@ sqlite/src/vdbeInt.h @@
+ If the order of members in struct Mem is changed, columnNullValue (defined in sqlite/vdbeapi.c) needs to be reordered accordingly.
@@ sqlite/src/vdbeaux.c @@
+ If a new fixed-length serial type is introduced, sqlite3IsFixedLengthSerialType (defined in sqlite/vdbeInt.h) needs to be changed accordingly.
@@ db/glue.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/appsock_handler.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/handle_buf.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/fdb_bend.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/prefault_helper.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/sqlglue.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ schemachange/sc_global.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ util/logmsg.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ util/walkback.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ berkdb/rep/rep_record.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ berkdb/btree/bt_rec.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ berkdb/dbreg/dbreg_util.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ csc2/macc_so.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/reqlog.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/glue.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/sqlinterfaces.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/osqlcheckboard.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/dohsql.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/eventlog.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/appsock_handler.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/trigger.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/fdb_boots.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/db_metrics.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/db_fingerprint.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/machclass.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/sqlglue.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ net/net.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ schemachange/sc_global.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/glue.c @@
+ Please replace any pthread_xxx() function in your diff with corresponding Pthread_xxx() wrapper.
```

File: plugins/newsql/newsql.c [hunk 1]
```diff
@@ -2210,6 +2217,24 @@ static CDB2QUERY *read_newsql_query(struct dbenv *dbenv,
         goto retry_read;
     }
 
+    if (gbl_uses_externalauth && !identity_dbinfo_sent &&
+        query->sqlquery && !query->sqlquery->identity) {
+        int client_supports_identity = 0;
+        for (int ii = 0; ii < query->sqlquery->n_features; ++ii) {
```
Comment (markhannum) [inline plugins/newsql/newsql.c:2223]: If the client doesn't support identity, does this loop always run?

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>6/356 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>verify_db_parallel_generated [core dumped]<br>updater_latency<br>sc_vutf8_validate<br>sc_vutf8_validate_off_osql_odh_blob_generated<br>datetime<br>logfill</p>

File: plugins/newsql/newsql.c (hunk 1 above)
Comment (mohitkhullar) [reply plugins/newsql/newsql.c:2223]: we don't  'goto retry_read' if client doesn't support the feature

Comment (mohitkhullar) [reply plugins/newsql/newsql.c:2223]: but we still check all the features, if server expects identity (uses_externalauth), but client hasn't sent it..

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ cdb2api/cdb2api.c @@
+ Please port the change to cdb2jdbc if necessary.
+ If you add any internal configuration state to this file, please update the reset_the_configuration() function as well to include it.
@@ db/db_tunables.h @@
+ Please update the tunable documentation (docs/pages/config/config_files.md) accordingly.
@@ sqlite/ext/comdb2/tables.c @@
+ If a new system table is added, be sure to document it in docs/pages/programming/system_tables.md.
@@ sqlite/src/parse.y @@
+ Custom tokens and rules must be defined after SQLite's.
+ Please update SQL documentation and bubble diagrams to reflect the changes you've made, if necessary.
@@ sqlite/src/vdbeInt.h @@
+ If the order of members in struct Mem is changed, columnNullValue (defined in sqlite/vdbeapi.c) needs to be reordered accordingly.
@@ sqlite/src/vdbeaux.c @@
+ If a new fixed-length serial type is introduced, sqlite3IsFixedLengthSerialType (defined in sqlite/vdbeInt.h) needs to be changed accordingly.
@@ db/glue.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/appsock_handler.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/handle_buf.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/fdb_bend.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/prefault_helper.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ db/sqlglue.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ schemachange/sc_global.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ util/logmsg.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ util/walkback.c @@
+ Please remove debugging statements, and replace fprintf/printf with logmsg.
@@ berkdb/rep/rep_record.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ berkdb/btree/bt_rec.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ berkdb/dbreg/dbreg_util.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ csc2/macc_so.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/reqlog.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/glue.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/sqlinterfaces.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/osqlcheckboard.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/dohsql.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/eventlog.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/appsock_handler.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/trigger.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/fdb_boots.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/db_metrics.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/db_fingerprint.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/machclass.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/sqlglue.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ net/net.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ schemachange/sc_global.c @@
+ Please make sure that a mutex is unlocked before returning.
@@ db/glue.c @@
+ Please replace any pthread_xxx() function in your diff with corresponding Pthread_xxx() wrapper.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>1/356 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>union_parallel</p>

**Comment by mohitkhullar** [comment]
Thanks

**Pull request merged**